applications/FreeSurfaceApplication/custom_utilities/edgebased_levelset_substep.h

```diff
@@ -51,2028 +51,2012 @@
+            //        }
+            //
+            //
+            //         KRATOS_WATCH("step1 before rk loop")
+            // 	KRATOS_WATCH(mnumsubsteps)
+            // 	KRATOS_WATCH(mPn)
+            // 	KRATOS_WATCH(mPn1)
+            // 	KRATOS_WATCH(mPi)
+            // 	KRATOS_WATCH(mvel_n1)
+            // 	KRATOS_WATCH(mvel_n)
 
 #ifdef DEBUG_OUTPUT
-        KRATOS_WATCH("before RK of step1 - new")
-        double aux_v=0.0;
-        for (int i_node = 0; i_node < mvel_n1.size(); i_node++)
-            aux_v += inner_prod(mvel_n1[i_node],mvel_n1[i_node]);
-        double aux_oldv=0.0;
-        for (int i_node = 0; i_node < mvel_n1.size(); i_node++)
-            aux_oldv += inner_prod(mvel_n[i_node],mvel_n[i_node]);
-        double aux_pi=0.0;
-        for (int i_node = 0; i_node < mvel_n1.size(); i_node++)
-            aux_pi += inner_prod(mPi[i_node],mPi[i_node]);
-
-        KRATOS_WATCH(inner_prod(mPn,mPn));
-        KRATOS_WATCH(aux_v);
-        KRATOS_WATCH(aux_oldv);
-        KRATOS_WATCH(aux_pi);
-        KRATOS_WATCH(inner_prod(mdistances,mdistances));
-        KRATOS_WATCH(inner_prod(mViscosity,mViscosity));
+            KRATOS_WATCH("before RK of step1 - new")
+            double aux_v = 0.0;
+            for (int i_node = 0; i_node < mvel_n1.size(); i_node++)
+                aux_v += inner_prod(mvel_n1[i_node], mvel_n1[i_node]);
+            double aux_oldv = 0.0;
+            for (int i_node = 0; i_node < mvel_n1.size(); i_node++)
+                aux_oldv += inner_prod(mvel_n[i_node], mvel_n[i_node]);
+            double aux_pi = 0.0;
+            for (int i_node = 0; i_node < mvel_n1.size(); i_node++)
+                aux_pi += inner_prod(mPi[i_node], mPi[i_node]);
+
+            KRATOS_WATCH(inner_prod(mPn, mPn));
+            KRATOS_WATCH(aux_v);
+            KRATOS_WATCH(aux_oldv);
+            KRATOS_WATCH(aux_pi);
+            KRATOS_WATCH(inner_prod(mdistances, mdistances));
+            KRATOS_WATCH(inner_prod(mViscosity, mViscosity));
 #endif
 
-        CalcVectorType auxn = mvel_n;
-
-        double n_substeps = mnumsubsteps+1;
+            CalcVectorType auxn = mvel_n;
 
-        double reduced_it = 0;
+            double n_substeps = mnumsubsteps + 1;
 
-        double energy_initial = 0.0;
-        double energy_final = 1.0;
+            double reduced_it = 0;
 
-        //compute initial kinetic energy
-        #pragma omp parallel for firstprivate(n_nodes) reduction(+:energy_initial)
-        for (int i_node = 0; i_node < n_nodes; i_node++)
-            if (mdistances[i_node] <= 0.0)
-                energy_initial += mr_matrix_container.GetLumpedMass()[i_node] * inner_prod(mvel_n[i_node],mvel_n[i_node]);
+            double energy_initial = 0.0;
+            double energy_final = 1.0;
 
-        //KRATOS_WATCH(energy_initial)
-        // KRATOS_WATCH(n_substeps)
+            // compute initial kinetic energy
+#pragma omp parallel for firstprivate(n_nodes) reduction(+ \
```

Comment:
In the future you can consider upgrading these to the `ParallelUtilities`.

Reply:
Thanks, I will do it in a future PR.